pkg/odo/cli/url/list.go

```diff
@@ -67,6 +69,23 @@ func (o *URLListOptions) Run() (err error) {
 	localUrls := o.localConfigInfo.GetUrl()
 
 	if log.IsJSON() {
+		if len(localUrls) != 0 {
+			for _, u := range localUrls {
```

Comment:
might be a good idea to extract this out into a function, this is becoming a bit less readable in my opinion

Reply:
:+1:

Reply:
maybe a more descriptive variable name than `u`

Reply:
This whole logic should be probably part of `url.List`
The `Url` struct should have the `Status` field, similar to how it is done for `Component` to indicate if the Url is pushed or not.